compiler/rustc_middle/src/ty/inhabitedness/inhabited_predicate.rs

```diff
@@ -190,10 +190,8 @@ impl<'tcx> InhabitedPredicate<'tcx> {
             (Self::And(&[a, b]), c) | (c, Self::And(&[a, b])) => {
                 if let Some(ac) = a.reduce_and(tcx, c) {
                     Some(ac.and(tcx, b))
-                } else if let Some(bc) = b.reduce_and(tcx, c) {
-                    Some(Self::And(tcx.arena.alloc([a, bc])))
                 } else {
-                    None
+                    b.reduce_and(tcx, c).map(|bc| Self::And(tcx.arena.alloc([a, bc])))
                 }
```

Comment:
I agree with the others, this feels less legible now than before. I won't block this though if you feel that's an improvement